internal/encryption/keyprovider/keyprovider.go

```diff
@@ -24,5 +27,5 @@ type Descriptor interface {
 
 type KeyProvider interface {
 	// Provide provides an encryption and decryption keys. If the process fails, it returns an error.
-	Provide() (Output, error)
+	Provide(KeyMeta) (Output, KeyMeta, error)
```

Comment:
I think @janosdebugs's idea was that the metadata is passed in via constructor?

No strong opinion on my side, but please make sure to discuss.

Reply:
@janosdebugs and I spent an hour going over different designs and settled on this one for now.